ios/imagemounter/personalized_image_mounter.go

```diff
@@ -0,0 +1,196 @@
+package imagemounter
+
+import (
+	"fmt"
+	"github.com/Masterminds/semver"
+	"github.com/danielpaulus/go-ios/ios"
+	log "github.com/sirupsen/logrus"
+	"io"
+	"os"
+	"path"
+)
+
+type PersonalizedDeveloperDiskImageMounter struct {
+	deviceConn ios.DeviceConnectionInterface
+	plistRw    ios.PlistCodecReadWriter
+	version    *semver.Version
+	tss        tssClient
+	ecid       uint64
+}
+
+func NewPersonalizedDeveloperDiskImageMounter(entry ios.DeviceEntry, version *semver.Version) (PersonalizedDeveloperDiskImageMounter, error) {
+	values, err := ios.GetValuesPlist(entry)
+	if err != nil {
+		return PersonalizedDeveloperDiskImageMounter{}, nil
+	}
+	var ecid uint64
+	if e, ok := values["UniqueChipID"].(uint64); ok {
+		ecid = e
+	} else {
+		return PersonalizedDeveloperDiskImageMounter{}, fmt.Errorf("could not get ECID from device")
```

Comment:
I want to start adding function names to errors so it will be easier to debug later. 
Can you add "NewPersonalizedDeveloperDiskImageMounter:" in front of the error please?

Reply:
sure, how do you want to handle function names that are present in many places like `Write`? Only the function name, or also the type if available?
I guess in most cases the name+message should be enough to get to the right place easily

Reply:
Good question, I think yeah the function name is probably enough. Also if we do this consistently from now on, you should be able to get something like a Java Stacktrace, so it should be relatively clear which call to Write it was from the other errrors wrapping this one.